### PR TITLE
Create remove update tasks

### PR DESCRIPTION
**Pull request by juanmanuelbyc**
Second part of this week project where we just worked on:

Remove all hardcoded items from the tasks array.
Create a new JavaScript file for the new functionality.
Implement a function for adding a new task (add a new element to the array).
Implement a function for deleting a task (remove an element from the array).
Implement a function for editing task descriptions.
By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/modules/tasksClass.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    const newTask = {
+      description: newTaskDescription.value,
+      completed: false,
+      index: this.tasks.length,
+    };
+    this.tasks.push(newTask);
+    this.clear();
+    this.setLocalStorage(this.tasks);
+    this.updateTasks();
+  };
```
Comment (TedLivist) [inline src/modules/tasksClass.js]: - [ ] When a task is added, its `index` property should be set to the value of the array length. So for example, when I add the first task the `index` should `1`, the second task the `index` should be `2`, so on and so forth. Kindly fix that.

  add-

File: src/modules/tasksClass.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  deleteTask = (i) => {
+    const filteredTasks = this.tasks.filter((task) => task !== this.tasks[i]);
+    this.setLocalStorage(filteredTasks);
+    this.updateTasks();
+  };
+
```
Comment (TedLivist) [inline src/modules/tasksClass.js:83]: - [ ] When a task is deleted, the `index` properties of the remaining tasks in the localStorage should be reset to reflect their positions in the array. So for example, if there are tasks with indexes of `1, 2, 3` and I delete the one with 2, I expect that the new index properties would be `1, 2`. Kindly implement that.

  int-1

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @juanmanuelbyc,

Good job so far!
✔️ Good documentation
✔️ Addition and deletion of tasks on the page is well implemented
✔️ Implementation of task task editing
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/tasksClass.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    this.tasks.push(newTask);
+    this.clear();
+    this.setLocalStorage(this.tasks);
+    this.updateTasks();
+  };
+
+  deleteTask = (i) => {
+    const filteredTasks = this.tasks.filter((task) => task !== this.tasks[i]);
+    this.setLocalStorage(filteredTasks);
+    this.updateTasks();
```
Comment (AnselemOdims) [inline src/modules/tasksClass.js:81]: - [ ] As pointed out by the previous reviewer, please ensure that when a todo item is deleted/removed it should update all remaining items' indexes, so they represent the current list order and are unique.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @juanmanuelbyc , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/modules/tasksClass.js (hunk 3 above)
Comment (juanmanuelbyc) [reply src/modules/tasksClass.js:81]: Hey Anselem! As you can see in the line 77 I call the function: this.updateTasks; 
Inside that function I update the indexes for remaining items.

Comment (juanmanuelbyc) [reply src/modules/tasksClass.js:81]: Hey Anselem!!! Are you there?

**Comment by juanmanuelbyc** [comment]
@AnselemOdims Anselem: I'm actually updating the tasks indexes successfully. Please double-check your review.

**Review by bimbolabuari — COMMENTED** [review]
Hello @juanmanuelbyc 👋,

_This is an additional review as a result of the second opinion request._

- Kudos for updating your array index as requested by the previous CR, your index update works as expected 👍 

As this was the only reason your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away :)

Keep rocking 🚀!

**Review by zhadier — APPROVED** [review]
## Hi @juanmanuelbyc  👋  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 



# Congratulations! 🎉 

✔️ No linter errors
✔️ Professional pull request title and description

### Keep Rocking!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**